### PR TITLE
Adding the Python-C AIHWKIT  package.

### PR DESCRIPTION
**Pull request by kkvtran**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kkvtran** [comment]
@conda-forge-admin, please ping team @conda-forge/staged-recipes

**Comment by kkvtran** [comment]
Please review my PR.  Thanks.

**Comment by kkvtran** [comment]
@conda-forge-admin, please ping team @conda-forge/staged-recipes
Please review the PR.  thanks.

**Comment by kkvtran** [comment]
@conda-forge/help-python ready for review!  Thanks.

**Comment by kkvtran** [comment]
@conda-forge-admin, please ping @conda-forge/help-python-c  Would you review this PR?  Thanks.

**Comment by kkvtran** [comment]
I agree to be listed as a maintainer for this package.

File: recipes/aihwkit/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+aihwkit_variant:
+  - cpu
+  - gpu
```
Comment (isuruf) [inline recipes/aihwkit/conda_build_config.yaml:3]: You don't need this. You can use `cuda_compiler_version == "None"` for the CPU version.

Comment (kkvtran) [reply recipes/aihwkit/conda_build_config.yaml]: Thank you for the review comment.  I removed the conda_build_config.yaml and updated the meta.yaml and build.sh accordingly.   Thanks again.

**Comment by kkvtran** [comment]
@conda-forge-admin, please ping @conda-forge/help-python-c 
Would you review this PR? Thanks.

File: recipes/aihwkit/conda_build_config.yaml (hunk 1 above)
Comment (kkvtran) [reply recipes/aihwkit/conda_build_config.yaml]: I pushed out the changes.  The PR was built successfully.  Would you please review the PR?  Thanks a million.

File: recipes/aihwkit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - mkl-include
+    - pip ==23.0.1
+    - scikit-build
+    - cudatoolkit  # [cuda_compiler_version != 'None']
+    - pytorch =*=cuda*   # [cuda_compiler_version != 'None']
+    - pytorch      # [cuda_compiler_version == 'None']
+  run:
+    - python
+    - mkl
+    - libgomp
```
Comment (isuruf) [inline recipes/aihwkit/meta.yaml:43]:
```suggestion
```

File: recipes/aihwkit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 1cfd8ca9fec90c7df1d0eacefbc5b421008bbf8cefda481793cd9e3d75710ae7
+
+build:
+  skip: true  # [win or osx]
+  skip: true  # [cuda_compiler_version in ("10.2", "11.0", "11.1")]
+  missing_dso_whitelist:
+    - $RPATH/libtorch_python.so
+    - lib/libgomp.so.1
```
Comment (isuruf) [inline recipes/aihwkit/meta.yaml:19]:
```suggestion
```

File: recipes/aihwkit/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != 'None']
+    - cmake
+    - numpy
+    - pybind11
+    - make
```
Comment (isuruf) [inline recipes/aihwkit/meta.yaml:30]:
```suggestion
    - make
    - libgomp   # [linux]
```

**Comment by kkvtran** [comment]
@conda-forge-admin, please ping @conda-forge/help-python-c
Would you review this PR? Thanks.

File: recipes/aihwkit/meta.yaml (hunk 4 above)
Comment (kkvtran) [reply recipes/aihwkit/meta.yaml:29]: I made the suggested change. Would you please review the PR?  thanks.

File: recipes/aihwkit/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    - libgomp   # [linux]
+  host:
+    - python
+    - mkl
```
Comment (isuruf) [inline recipes/aihwkit/meta.yaml:33]:
```suggestion
    - mkl
    - libblas =*=*_mkl
```

File: recipes/aihwkit/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    - mkl-include
+    - pip ==23.0.1
+    - scikit-build
+    - cudatoolkit  # [cuda_compiler_version != 'None']
+    - pytorch =*=cuda*   # [cuda_compiler_version != 'None']
+    - pytorch      # [cuda_compiler_version == 'None']
+  run:
+    - python
+    - mkl
```
Comment (isuruf) [inline recipes/aihwkit/meta.yaml:42]:
```suggestion
    - mkl
    - libblas =*=*_mkl
```

**Pull request merged**